kube-scheduler panic in NodesHaveSameTopologyKey

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
We are facing regular kube-scheduler panics
**What you expected to happen**:
kube-scheduler should not panic
**How to reproduce it (as minimally and precisely as possible)**:
Sorry. I haven't been able to reproduce it, it started happening after we upgraded to 1.16.6. 
**Anything else we need to know?**:
panic logs 
```sh
E0323 14:55:21.352508 1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)goroutine 369514 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1810be0, 0x2b37e00)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
panic(0x1810be0, 0x2b37e00)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util/topologies.go:58
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util.NodesHaveSameTopologyKey(...)
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*podAffinityPriorityMap).processTerm(0xc0014cf620, 0xc000be8d08, 0xc00143b400, 0xc000ffbc00, 0x0, 0xffffffffffffff9c, 0xc00027a6a8, 0x0)
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*podAffinityPriorityMap).processTerms(0xc0014cf620, 0xc000be8d00, 0x1, 0x1, 0xc00143b400, 0xc000ffbc00, 0x0, 0xffffffffffffffff, 0x0, 0x0)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:76 +0x135
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:87 +0xa0
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*InterPodAffinity).CalculateInterPodAffinityPriority.func1(0xc000ffbc00, 0x0, 0x0)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:141 +0x407
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*InterPodAffinity).CalculateInterPodAffinityPriority.func2(0xb)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:190 +0x113
k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil.func1(0xc001387ae0, 0xc000a3e0d0, 0xc0000ad120, 0xc000dff540)
created by k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:57 +0xbf
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:49 +0x148
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x15ec535]
panic: runtime error: invalid memory address or nil pointer dereference
goroutine 369514 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1810be0, 0x2b37e00)
/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util.NodesHaveSameTopologyKey(...)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util/topologies.go:58
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*podAffinityPriorityMap).processTerm(0xc0014cf620, 0xc000be8d08, 0xc00143b400, 0xc000ffbc00, 0x0, 0xffffffffffffff9c, 0xc00027a6a8, 0x0)
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*podAffinityPriorityMap).processTerms(0xc0014cf620, 0xc000be8d00, 0x1, 0x1, 0xc00143b400, 0xc000ffbc00, 0x0, 0xffffffffffffffff, 0x0, 0x0)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:76 +0x135
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*InterPodAffinity).CalculateInterPodAffinityPriority.func1(0xc000ffbc00, 0x0, 0x0)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:141 +0x407
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:87 +0xa0
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/interpod_affinity.go:190 +0x113
k8s.io/kubernetes/pkg/scheduler/algorithm/priorities.(*InterPodAffinity).CalculateInterPodAffinityPriority.func2(0xb)
k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil.func1(0xc001387ae0, 0xc000a3e0d0, 0xc0000ad120, 0xc000dff540)
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:57 +0xbf
created by k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil
/workspace/anago-v1.16.6-beta.0.3+72c30166b2105c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:49 +0x148
I0323 14:55:22.339423 1 serving.go:319] Generated self-signed cert in-memory
I0323 14:55:22.824664 1 server.go:148] Version: v1.16.6
I0323 14:55:22.825689 1 defaults.go:91] TaintNodesByCondition is enabled, PodToleratesNodeTaints predicate is mandatory
W0323 14:55:22.834286 1 authorization.go:47] Authorization is disabled
I0323 14:55:22.834347 1 deprecated_insecure_serving.go:51] Serving healthz insecurely on [::]:10251
W0323 14:55:22.834298 1 authentication.go:79] Authentication is disabled
I0323 14:55:22.834946 1 secure_serving.go:123] Serving securely on 127.0.0.1:10259
I0323 14:55:22.935620 1 leaderelection.go:241] attempting to acquire leader lease kube-system/kube-scheduler...
I0323 14:55:38.533050 1 leaderelection.go:251] successfully acquired lease kube-system/kube-scheduler
E0323 14:56:01.874600 1 factory.go:585] pod is already present in the activeQ
E0323 14:56:21.620005 1 factory.go:585] pod is already present in the activeQ
E0323 14:56:47.490600 1 factory.go:585] pod is already present in the activeQ
```
**Environment**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:31:31Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:23:21Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: `AWS`
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="16.04.6 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.6 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
- Kernel (e.g. `uname -a`):
```
Linux ip-X-X-X-X.X-X-X.compute.internal 4.4.0-1066-aws #76-Ubuntu SMP Thu Aug 16 16:21:21 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug): `weave-net:2.6.0`
- Others:

--- a/pkg/scheduler/algorithm/predicates/predicates.go
+++ b/pkg/scheduler/algorithm/predicates/predicates.go
@@ -1380,8 +1380,6 @@ func (c *PodAffinityChecker) getMatchingAntiAffinityTopologyPairsOfPods(pod *v1.
 		existingPodNode, err := c.info.GetNodeInfo(existingPod.Spec.NodeName)
 		if err != nil {
 			if apierrors.IsNotFound(err) {
-				klog.Errorf("Pod %s has NodeName %q but node is not found",
-					podName(existingPod), existingPod.Spec.NodeName)
 				continue
 			}
 			return nil, err

--- a/pkg/scheduler/algorithm/priorities/interpod_affinity.go
+++ b/pkg/scheduler/algorithm/priorities/interpod_affinity.go
@@ -115,7 +115,6 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 		existingPodNode, err := ipa.info.GetNodeInfo(existingPod.Spec.NodeName)
 		if err != nil {
 			if apierrors.IsNotFound(err) {
-				klog.Errorf("Node not found, %v", existingPod.Spec.NodeName)
 				return nil
 			}
 			return err

--- a/pkg/scheduler/internal/cache/BUILD
+++ b/pkg/scheduler/internal/cache/BUILD
@@ -16,6 +16,7 @@ go_library(
         "//pkg/util/node:go_default_library",
         "//staging/src/k8s.io/api/core/v1:go_default_library",
         "//staging/src/k8s.io/api/storage/v1beta1:go_default_library",
+        "//staging/src/k8s.io/apimachinery/pkg/api/errors:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/labels:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/util/sets:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/util/wait:go_default_library",

--- a/pkg/scheduler/internal/cache/cache.go
+++ b/pkg/scheduler/internal/cache/cache.go
@@ -23,6 +23,7 @@ import (
 
 	v1 "k8s.io/api/core/v1"
 	storagev1beta1 "k8s.io/api/storage/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
@@ -698,8 +699,8 @@ func (cache *schedulerCache) GetNodeInfo(nodeName string) (*v1.Node, error) {
 	defer cache.mu.RUnlock()
 
 	n, ok := cache.nodes[nodeName]
-	if !ok {
-		return nil, fmt.Errorf("node %q not found in cache", nodeName)
+	if !ok || n.info.Node() == nil {
+		return nil, errors.NewNotFound(v1.Resource("node"), nodeName)
 	}
 
 	return n.info.Node(), nil

